### DISCUSSION TITLE
Lesson 12:  testCanDepositCollateralAndGetAccountInfo = Fail. Reason: Index out of Bounds

### DISCUSSION THREAD

**Discussion by CryptoNube** [Cyfrin/foundry-full-course-cu#1127] [Q&A] [answered]
Hi devs.
At 3:09:08, https://youtu.be/wUjYK5gwNZs?si=Mkjtw-prjpuVI_xT&t=11348, 
I get the following error:
![OutOfBounds](https://github.com/Cyfrin/foundry-full-course-f23/assets/83887025/098f5134-ca3f-4a58-bcd4-60c2df9915b3)
I can see that the error is with the `getAccountInformation`  function but, I do not know how to resolve.
Here is my repo:
https://github.com/CryptoNube/foundry-defi-stablecoin/tree/main

**Comment by 0xusmanf** [comment, marked as answer]
`i` in this [loop](https://github.com/CryptoNube/foundry-defi-stablecoin/blob/fa86f50c80c88721f5832cc88e0d5a565b77b272/src/DSCEngine.sol#L340) should only be less then `s_collateralTokens.length` instead of less then equal to `s_collateralTokens.length` which is causing the revert.

**Reply by CryptoNube** [reply]
Oh geez! I missed that.
Thank you very much!

**Reply by 0xusmanf** [reply]
You are welcome, happy to help mate!
